packages/contracts/src/diamond/libraries/LibStakingFormulas.sol

```diff
@@ -0,0 +1,192 @@
+        return _amount;
+    }
+
+    /// @dev formula Governance Rights corresponding to a staking shares LP amount
+    /// @param _stake , staking share
+    /// @param _amount , amount of LP tokens
+    /// @notice shares = (stake.shares * _amount )  / stake.lpAmount ;
+    function sharesForLP(
+        StakingShare.Stake memory _stake,
+        uint256[2] memory _shareInfo,
+        uint256 _amount
+    ) internal pure returns (uint256 _uLP) {
+        bytes16 a = _shareInfo[0].fromUInt(); // shares amount
+        bytes16 v = _amount.fromUInt();
+        bytes16 t = _stake.lpAmount.fromUInt();
+
+        _uLP = a.mul(v).div(t).toUInt();
+    }
+
+    /// @dev formula may add a decreasing rewards if locking end is near when removing liquidity
+    /// @param _stake , staking share
+    /// @param _amount , amount of LP tokens
+    /// @notice rewards = _amount;
+    // solhint-disable-block  no-unused-vars
+    /* solhint-disable no-unused-vars */
+    function lpRewardsRemoveLiquidityNormalization(
+        StakingShare.Stake memory _stake,
+        uint256[2] memory _shareInfo,
+        uint256 _amount
+    ) internal pure returns (uint256) {
+        return _amount;
+    }
+
+    /* solhint-enable no-unused-vars */
+    /// @dev formula may add a decreasing rewards if locking end is near when adding liquidity
+    /// @param _stake , staking share
+    /// @param _amount , amount of LP tokens
+    /// @notice rewards = _amount;
+    // solhint-disable-block  no-unused-vars
+    /* solhint-disable no-unused-vars */
+    function lpRewardsAddLiquidityNormalization(
+        StakingShare.Stake memory _stake,
+        uint256[2] memory _shareInfo,
+        uint256 _amount
+    ) internal pure returns (uint256) {
+        return _amount;
+    }
+
+    /// @dev formula duration multiply
+    /// @param _uLP , amount of LP tokens
+    /// @param _weeks , minimum duration of staking period
+    /// @param _multiplier , staking discount multiplier = 0.0001
+    /// @return _shares , amount of shares
+    /// @notice _shares = (1 + _multiplier * _weeks^3/2) * _uLP
+    //          D32 = D^3/2
+    //          S = m * D32 * A + A
+    function durationMultiply(
+        uint256 _uLP,
+        uint256 _weeks,
+        uint256 _multiplier
+    ) internal pure returns (uint256 _shares) {
+        bytes16 unit = uint256(1 ether).fromUInt();
+        bytes16 d = _weeks.fromUInt();
+        bytes16 d32 = (d.mul(d).mul(d)).sqrt();
+        bytes16 m = _multiplier.fromUInt().div(unit); // 0.0001
+        bytes16 a = _uLP.fromUInt();
+
+        _shares = m.mul(d32).mul(a).add(a).toUInt();
+    }
+
+    /// @dev formula bonding
+    /// @param _shares , amount of shares
+    /// @param _currentShareValue , current share value
+    /// @param _targetPrice , target Ubiquity Dollar price
+    /// @return _uBOND , amount of bonding shares
+    /// @notice UBOND = _shares / _currentShareValue * _targetPrice
+    // newShares = A / V * T
+    function bonding(
```

Comment:
```suggestion
    function staking(
```

We renamed Bonding to staking. Also uBOND refers to the deprecated credit concept (its just a CreditNFT with no expiration basically.)

You should refer to the "bonding shares" as StakingTokens I think thats less ambiguous.

Reply:
Depending on availability I might just open a new PR to handle renames again.

Reply:
ok I will rename